### ISSUE TITLE
Does TensorFlow support temporal convolution

### ISSUE THREAD

**Issue by zer0n** [tensorflow/tensorflow#166]
like [Torch does](https://github.com/torch/nn/blob/master/doc/convolution.md)? I assume not, hence asking here.

Temporal convolution is useful for encoding a sentence. I find it work better than RNN/LSTM.

**Comment by jorenvs** [comment]
+1

**Comment by rafaljozefowicz** [comment]
Could you tell me what's the difference between the temporal convolution and using tf.nn.conv_2d with a very specific kernel, e.g. [height=kW, width=embedding_size, in=inputFrameSize, out=outputFrameSize]?

**Comment by rafaljozefowicz** [comment]
Related discussion on reddit: https://www.reddit.com/r/MachineLearning/comments/3sjl7o/tensorflow_vs_theano_vs_torch_comparison/cwxsq08

**Comment by zer0n** [comment]
In Theano's [conv2d](http://deeplearning.net/software/theano/library/tensor/nnet/conv.html#theano.tensor.nnet.conv.conv2d), you can specify the `image_shape` to be `None`, which signals that it's not a constant.

It's unclear if non-const input size is supported in TensorFlow's `conv2d`.

**Comment by vrv** [comment]
Only strides / padding are constants, input and filter can be dynamic sizes: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/api_docs/python/nn.md#conv2d

**Comment by zer0n** [comment]
Thanks.
